src/main/java/net/minecraftforge/items/wrapper/EntityEquipmentInvWrapper.java

```diff
@@ -104,12 +104,9 @@ public void clearInv()
     }
 
     @Override
-    @Nonnull
-    public ItemStack setStackInSlot(int slot, @Nonnull ItemStack stack)
+    public void setStackInSlot(int slot, @Nonnull ItemStack stack)
```

Comment:
Returning the previous stack is actually pretty useful.

One of the main issues with using `IItemHandler` for GUI interactions, is that there is no atomic way to swap a stack. Only use extract and insert and hope it does not get messed up somehow. Or us the `IItemHandlerModifiable`, which also has it's design issues. Like some helpers doing unchecked casts for it, Causing crashes or at least no functionality when used like documented.

Yes, you can combine `getStackInSlot` or similar, but this is stil not automatic. There are no guarantees that the handler does not modify the stack after using `setStackInSlot`, so it requires and explicit copy each time. Just returning the previous stack (with out any reference to some internal stack), would make swapping a slot very trivial. Pretty much no needs for safeguards. And it can be even used for fakeslots by simply returning `ItemStack.EMPTY` without any additional safeguards.

Reply:
setStackInSlot is in IItemHandlerModifiable